### PR TITLE
Highlight resources with critical licenses

### LINKED ISSUE #1028: Resources with critical licenses are highlighted
The icons indicating presence of signals shall be colored based on highest criticality for resources with critical signals and also their tooltip texts should be modified:
Screenshot 2022-09-28 at 11 34 38

Part of #1012

### PR DESCRIPTION
**Pull request by vasily-pozdnyakov**
### Summary of changes

The icons indicating presence of signals are colored based on highest criticality for resources with critical signals and also their tooltip texts are modified.
Fix #1028
Part of #1012

**Before:**
image

**After:**
image

### CODE REVIEW

File: src/Frontend/Components/Icons/Icons.tsx [hunk 1]
```diff
@@ -102,12 +107,26 @@ export function FollowUpIcon(props: IconProps): ReactElement {
   );
 }
 
-export function SignalIcon(): ReactElement {
+export function SignalIcon(props: SignalIconProps): ReactElement {
+  const tooltipText =
+    props.criticality === Criticality.High
+      ? 'has highly critical signals'
```
Comment (MarkusObendrauf) [inline src/Frontend/Components/Icons/Icons.tsx:113]: imo 'has high criticality signals' would be clearer here (and below: 'has medium criticality signals')

File: src/Frontend/Components/Icons/Icons.tsx [hunk 2]
```diff
@@ -102,12 +107,26 @@ export function FollowUpIcon(props: IconProps): ReactElement {
   );
 }
 
-export function SignalIcon(): ReactElement {
+export function SignalIcon(props: SignalIconProps): ReactElement {
+  const tooltipText =
+    props.criticality === Criticality.High
```
Comment (leslielazzarino) [inline src/Frontend/Components/Icons/Icons.tsx:112]: I would replace this if with an object, it would be cleaner.

File: src/Frontend/Components/Icons/Icons.tsx [hunk 3]
```diff
@@ -102,12 +107,26 @@ export function FollowUpIcon(props: IconProps): ReactElement {
...
   return (
-    <MuiTooltip sx={classes.tooltip} title="has signals">
+    <MuiTooltip sx={classes.tooltip} title={tooltipText}>
       <AnnouncementIcon
         aria-label={'Signal icon'}
-        sx={classes.nonClickableIcon}
+        sx={{
+          ...baseIcon,
+          color:
+            props.criticality === Criticality.High
```
Comment (leslielazzarino) [inline src/Frontend/Components/Icons/Icons.tsx:124]: Same here.

File: src/Frontend/Components/ResourceBrowser/get-tree-item-label.tsx [hunk 4]
```diff
@@ -75,6 +83,27 @@ export function getTreeItemLabel(
...
 
+function getCriticality(
+  nodeId: string,
+  resourcesToExternalAttributions: ResourcesToAttributions,
+  externalAttributions: Attributions
+): string | undefined {
+  if (hasExternalAttribution(nodeId, resourcesToExternalAttributions)) {
+    let criticality;
+    const attributionsForResource = resourcesToExternalAttributions[nodeId];
+    for (const attribution of attributionsForResource) {
```
Comment (MarkusObendrauf) [inline src/Frontend/Components/ResourceBrowser/get-tree-item-label.tsx:94]: I find this a bit hard to follow. Suggestion:
```
for (const attribution of attributionsForResource) {
  if (externalAttributions[attribution].criticality === Criticality.High) {
    return Criticality.High
  }
}
```
followed by a similar loop for medium

**Review by MarkusObendrauf — CHANGES_REQUESTED** [review]

File: src/Frontend/Components/ResourceBrowser/get-tree-item-label.tsx [hunk 5]
```diff
@@ -75,6 +83,27 @@ export function getTreeItemLabel(
   );
 }
 
+function getCriticality(
```
Comment (leslielazzarino) [inline src/Frontend/Components/ResourceBrowser/get-tree-item-label.tsx:86]: This should be tested thoroughly.

File: src/Frontend/Components/Icons/Icons.tsx [hunk 6]
```diff
@@ -102,12 +107,26 @@ export function FollowUpIcon(props: IconProps): ReactElement {
   );
 }
 
-export function SignalIcon(): ReactElement {
+export function SignalIcon(props: SignalIconProps): ReactElement {
```
Comment (leslielazzarino) [inline src/Frontend/Components/Icons/Icons.tsx:110]: The behavior of this component should be tested.

**Review by leslielazzarino — CHANGES_REQUESTED** [review]

File: src/Frontend/Components/ResourceBrowser/get-tree-item-label.tsx (hunk 4 above)
Comment (vasily-pozdnyakov) [reply src/Frontend/Components/ResourceBrowser/get-tree-item-label.tsx:94]: I got your point now, will do it.

**Review by MarkusObendrauf — APPROVED** [review]

**Review by leslielazzarino — APPROVED** [review]
